components/hab/src/license.rs

```diff
@@ -0,0 +1,122 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{common::ui::{self,
+                         UIReader,
+                         UIWriter,
+                         UI},
+            error::{Error,
+                    Result},
+            hcore::{fs::{am_i_root,
+                         FS_ROOT_PATH},
+                    users::get_current_username}};
+use chrono::prelude::*;
+use dirs;
+use serde_yaml;
+use std::{fs::{self,
+               File},
+          io::Write,
+          path::PathBuf};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct LicenseData {
+    pub date_accepted: DateTime<Utc>,
+    pub accepting_product: String,
+    pub accepting_product_version: String,
+    pub user: Option<String>,
+    pub file_format: String,
+}
+
+impl LicenseData {
+    pub fn new() -> Self {
+        LicenseData { date_accepted: Utc::now(),
+                      accepting_product: String::from("hab"),
+                      accepting_product_version: super::VERSION.to_string(),
+                      user: get_current_username(),
+                      file_format: String::from("1.0"), }
+    }
+}
+
+pub fn check_for_license_acceptance_and_prompt(ui: &mut UI) -> Result<()> {
+    if license_exists() {
+        return Ok(());
+    }
+
+    ui.heading("+---------------------------------------------+")?;
+    ui.heading("            Chef License Acceptance")?;
+    ui.br()?;
+    ui.info("Before you can continue, 1 product license must be accepted.")?;
+    ui.info("View the license at https://www.chef.io/end-user-license-agreement")?;
+    ui.br()?;
+    ui.info("License that needs accepting:")?;
+    ui.br()?;
+    ui.info("  * Habitat")?;
+    ui.br()?;
+
+    if ui.prompt_yes_no("Do you accept the 1 product license?", Some(false))? {
+        accept_license(ui)
+    } else {
+        ui.br()?;
+        ui.info("If you do not accept this license you will not be able to use Chef products.")?;
+        ui.br()?;
+
+        if ui.prompt_yes_no("Do you accept the 1 product license?", Some(false))? {
+            accept_license(ui)
+        } else {
+            ui.br()?;
+            ui.heading("+---------------------------------------------+")?;
+            Err(Error::LicenseNotAccepted)
+        }
+    }
+}
+
+pub fn accept_license(ui: &mut UI) -> Result<()> {
+    if license_exists() {
+        ui.info("You have already accepted the license.")?;
+        return Ok(());
+    }
+
+    ui.br()?;
+    ui.info("Accepting 1 product license...")?;
+
+    let license = LicenseData::new();
+    let content = serde_yaml::to_string(&license)?;
+    fs::create_dir_all(license_path())?;
+    let mut file = File::create(license_file())?;
+    file.write_all(content.as_bytes())?;
+
+    ui.status(ui::Status::Custom(ui::Glyph::CheckMark, String::from("")),
```

Comment:
What's the empty string for?

Reply:
What I want to display is `CHECKMARK_GLYPH_HERE 1 product license accepted.` but the way our ui library sets this stuff up, each glyph has to have some text next to it.  So I either need an empty string here, or pass an empty string to the `ui.status()` call to get the output I need.  This felt like a better place.  The output here btw is conforming to the license acceptance spec that chefdk and all the rest of our tools follow, so it's not something I had control over.

Reply:
The fact that this conforms to an external spec would be great to document.

It would also be nice to add a test that could ensure that we're conforming.

Reply:
I will link to the repo containing the spec in the module level documentation, but it's not the kind of thing that's testable in an automated way.